doc/source/authentication.rst

```diff
@@ -208,6 +208,28 @@ and obtain the confidential client credentials.
          userdata_params: {'state': 'state'}
          username_key: preferred_username
 
+Auth0
+^^^^^^^^^^^^^^
```

Comment:
Thanks for the PR @philvarner. A few comments...

Reply:
In rst, the underline should be the same length as the line above.

Reply:
> In rst, the underline should be the same length as the line above.

fixed